src/InputGroup/index.js

```diff
@@ -156,126 +164,145 @@ StyledInputGroup.defaultProps = {
   theme: defaultTheme,
 };
 
-class InputGroup extends React.PureComponent<Props, State> {
-  state = {
-    active: false,
-    filled: false,
-  };
-
-  inputID: string = randomID("inputGroupID");
-
-  componentDidMount() {
-    this.isFilled();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (this.props.children !== prevProps.children) {
-      this.isFilled();
-    }
-  }
-
-  isFilled = () =>
-    this.setState({
-      filled: !React.Children.map(
-        this.props.children,
-        child => child.props.value !== undefined && child.props.value !== "",
-      ).includes(false),
-    });
-
-  handleFocus = (ev: SyntheticInputEvent<HTMLInputElement>) => {
-    const { onFocus } = this.props;
-
-    this.setState({ active: true });
-    if (onFocus) {
-      onFocus(ev);
-    }
-  };
-
-  handleBlur = (ev: SyntheticInputEvent<HTMLInputElement>) => {
-    const { onBlur } = this.props;
-    this.isFilled();
-
-    this.setState({ active: false });
-    if (onBlur) {
-      onBlur(ev);
-    }
-  };
-
-  handleChange = (ev: SyntheticInputEvent<HTMLInputElement>) => {
-    const { onChange } = this.props;
-    this.isFilled();
-
-    if (onChange) {
-      onChange(ev);
-    }
-  };
-
-  render() {
-    const {
-      children,
-      label,
-      flex = "0 1 auto",
-      size = SIZE_OPTIONS.NORMAL,
-      help,
-      error,
-      dataTest,
-      spaceAfter,
-    } = this.props;
-
-    const { active, filled } = this.state;
+const findPropInChild = (propToFind, children) => {
+  return React.Children.map(children, el => {
+    if (el.props && el.props[propToFind]) return el.props[propToFind];
+    return null;
+  }).filter(el => el != null);
```

Comment:
if you are trying to get rid of all falsy values including false consider `filter(Boolean)` :)